framework/auth/views.py

```diff
@@ -51,88 +54,126 @@ def reset_password(auth, **kwargs):
         }
         raise HTTPError(400, data=error_data)
 
-    if request.method == 'POST' and form.validate():
-        # new random verification key, allows CAS to authenticate the user w/o password one time only.
-        user_obj.verification_key = security.random_string(20)
-        user_obj.set_password(form.password.data)
-        user_obj.save()
-        status.push_status_message('Password reset', kind='success', trust=False)
-        # Redirect to CAS and authenticate the user with a verification key.
-        return redirect(cas.get_login_url(
-            web_url_for('user_account', _absolute=True),
-            auto=True,
-            username=user_obj.username,
-            verification_key=user_obj.verification_key
-        ))
-
-    forms.push_errors_to_status(form.errors)
+    if request.method == 'GET':
+        pass
+
+    if request.method == 'POST':
+        if form.validate():
+            # new random verification key, allows CAS to authenticate the user w/o password, one-time only.
+            # this overwrite also invalidates the verification key generated by forgot_password_post
+            user_obj.verification_key = generate_verification_key()
+            user_obj.set_password(form.password.data)
+            user_obj.save()
+            status.push_status_message('Password reset', kind='success', trust=False)
+            # redirect to CAS and authenticate the user with the one-time verification key.
+            return redirect(cas.get_login_url(
+                web_url_for('user_account', _absolute=True),
+                username=user_obj.username,
+                verification_key=user_obj.verification_key
+            ))
+        else:
+            forms.push_errors_to_status(form.errors)
+            # Don't go anywhere
+
     return {
-        'verification_key': verification_key,
+        'verification_key': verification_key
```

Comment:
should have a trailing comma (https://www.python.org/dev/peps/pep-0008/#indentation - bottom)